eclipse-collections/src/main/java/org/eclipse/collections/impl/map/ordered/mutable/OrderedMapAdapter.java

```diff
@@ -0,0 +1,809 @@
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        return this.delegate.equals(o);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return this.delegate.hashCode();
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> newEmpty()
+    {
+        return OrderedMapAdapter.adapt(new LinkedHashMap<>());
+    }
+
+    @Override
+    public void forEachKeyValue(Procedure2<? super K, ? super V> procedure)
+    {
+        MapIterate.forEachKeyValue(this.delegate, procedure);
+    }
+
+    @Override
+    public boolean containsKey(Object key)
+    {
+        return this.delegate.containsKey(key);
+    }
+
+    @Override
+    public boolean containsValue(Object value)
+    {
+        return this.delegate.containsValue(value);
+    }
+
+    @Override
+    public int size()
+    {
+        return this.delegate.size();
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return this.delegate.isEmpty();
+    }
+
+    @Override
+    public Iterator<V> iterator()
+    {
+        return this.delegate.values().iterator();
+    }
+
+    @Override
+    public V remove(Object key)
+    {
+        return this.delegate.remove(key);
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> map)
+    {
+        this.delegate.putAll(map);
+    }
+
+    @Override
+    public MutableCollection<V> values()
+    {
+        return CollectionAdapter.adapt(this.delegate.values());
+    }
+
+    // TODO: OrderedSet
+    @Override
+    public MutableSet<Entry<K, V>> entrySet()
+    {
+        return SetAdapter.adapt(this.delegate.entrySet());
+    }
+
+    // TODO: OrderedSet
+    @Override
+    public MutableSet<K> keySet()
+    {
+        return SetAdapter.adapt(this.delegate.keySet());
+    }
+
+    @Override
+    public void clear()
+    {
+        this.delegate.clear();
+    }
+
+    @Override
+    public V get(Object key)
+    {
+        return this.delegate.get(key);
+    }
+
+    @Override
+    public V put(K key, V value)
+    {
+        return this.delegate.put(key, value);
+    }
+
+    @Override
+    public V removeKey(K key)
+    {
+        return this.delegate.remove(key);
+    }
+
+    @Override
+    public String toString()
+    {
+        return this.delegate.toString();
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> clone()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> toReversed()
+    {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".toReversed() not implemented yet");
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> take(int count)
+    {
+        if (count < 0)
+        {
+            throw new IllegalArgumentException("Count must be greater than zero, but was: " + count);
+        }
+
+        MutableOrderedMap<K, V> output = this.newEmpty();
```

Comment:
This implementation is consistent with TreeSortedMap. That's a type that cannot be pre-sized though. In general the abstract maps aren't as mature as the abstract collections, and can't even be fixed until a major version. I suppose this will be consistent in all transformation methods.

Reply:
May be we should then fix it for OrderedMapAdapter in 9.2.0 since it is a new implementation any case. In 10.0.0 seems like we have a ton of work to reduce the tech debt. What do you think?